File: Python Banking System/CheckingAccount.py
from BankAccount import BankAccount
class CheckingAccount(BankAccount):
    def __init__(self, accountNumber, accountHolderName, balance, bankType):
        super().__init__(accountNumber, accountHolderName, balance)
        self._bankType = bankType

    def displayAccountDetails(self):
        print("Displaying information...")
        print(f"Account Number: {self._accountNumber}")
        print(f"Account Name: {self._accountHolderName}")
        print(f"Account Balance: {self._balance}")
        print(f"Bank Type: {self._bankType}")
        print()

    def transferMoney(self, targetAccount, amount):
        if isinstance(targetAccount, CheckingAccount):
            if (self._balance - amount >= 0):
                self._balance -= amount
                targetAccount._balance += amount
                print("Transfer success")
            else:
                print("Insufficient funds to conduct transfer")
        else:
            print("Target account is not checking account, cannot transfer money")

    def atmWithdraw(self, amount, atmType):
        assert atmType != "", "ATM type cannot be empty"
        if atmType != self._bankType and atmType != "Exception": 
            confirm = input("Fee incurred of 20$, continue with withdrawal? (Y/N)")
            match (confirm):
                case "Y":
                    if (self._balance - (amount+20) >= 0):
                        self._balance -= (amount + 20)
                    else:
                        print("Insufficient funds to conduct withdrawal and pay fee")
                case "N":
                    print("Withdraw cancelled.")
                case default:
                    print("Unsupported action, please choose either Y or N")
            while confirm != "Y" and confirm != "N":
                confirm = input("Fee incurred of 20$, continue with withdrawal? (Y/N)")
                match (confirm):
                    case "Y":
                        if (self._balance - (amount+20) >= 0):
                            self._balance -= (amount + 20)
                        else:
                            print("Insufficient funds to conduct withdrawal and pay fee")
                    case "N":
                        print("Withdraw cancelled.")
                    case default:
                        print("Unsupported action, please choose either Y or N")
        else:
            print("Matching back type or exception allowed, no fees incurred...")
            if (self._balance - amount >= 0):
                self._balance -= amount
            else:
                print("Insufficient funds to conduct withdrawal and pay fee")

File: Java Banking System/demo/src/main/java/com/example/CheckingAccount.java
package com.example;
import java.util.Scanner;

public class CheckingAccount extends BankAccount{

    private String bankType;
    public CheckingAccount(int accountNumber, String accountHolderName, int balance, String bankType) {
        super(accountNumber, accountHolderName, balance);
        this.bankType = bankType;
    } 

    @Override
    public void displayAccountDetails(){
        System.out.println("Displaying information...");
        System.out.println("Account Number: " + this.getAccountNumber());
        System.out.println("Account Name: " + this.getAccountHolderName());
        System.out.println("Account Balance: " + this.getBalance());
        System.out.println("Bank Type: " + this.bankType + "\n");
    }
    
    public void transferMoney(CheckingAccount targetAccount, int amount){
        if (this.getBalance() - amount >= 0) {
            int subtractedBalance = this.getBalance() - amount;
            int addedBalance = targetAccount.getBalance() + amount;
            targetAccount.setBalance(addedBalance);
            this.setBalance(subtractedBalance);
        }
        else {
            System.out.println("Insufficient funds to conduct transfer");
        }
    }

    public void atmWithdraw(int amount, String atmType) {
        if ((atmType.compareTo(this.bankType) != 0) && (atmType.compareTo("Exception") != 0)){
            Scanner scanner = new Scanner(System.in);
            System.out.println("Fee incurred of 20$, continue with withdrawal? (Y/N)");
            String confirm = scanner.nextLine();
            switch (confirm){
                case "Y":
                    if (this.getBalance() - (amount+20) >= 0) {
                        this.setBalance(this.getBalance() - (amount+20));
                    }
                    else {
                        System.out.println("Insufficient funds to conduct withdrawal and pay fee");
                    }
                    break;
                case "N":
                    System.out.println("Withdraw cancelled.");
                    break;
                default:
                System.out.println("Unsupported action, please choose either Y or N");
            }
            while (confirm.compareTo("Y") != 0 && confirm.compareTo("N") != 0) {
                    System.out.println("Fee incurred of 20$, continue with withdrawal? (Y/N)");
                    confirm = scanner.nextLine();
                    switch (confirm){
                        case "Y":
                            if (this.getBalance() - (amount+20) >= 0) {
                                this.setBalance(this.getBalance() - (amount+20));
                            }
                            else {
                                System.out.println("Insufficient funds to conduct withdrawal and pay fee");
                            }
                            break;
                        case "N":
                            System.out.println("Withdraw cancelled.");
                            break;
                        default:
                        System.out.println("Unsupported action, please choose either Y or N");
                    }
            }
            scanner.close();
        }
        else {
            System.out.println("Matching back type or exception allowed, no fees incurred...");
            if (this.getBalance() - amount >= 0) {
                this.setBalance(this.getBalance() - amount);
            }
            else {
                System.out.println("Insufficient funds to conduct withdrawal and pay fee");
            }
        }
    }

    public String getBankType(){
        return this.bankType;
    }
}
